src/ppx_import.cppo.ml

```diff
@@ -227,17 +239,26 @@ let type_declaration mapper type_decl =
     | PTyp ({ ptyp_desc = Ptyp_constr ({ txt = lid; loc }, _) } as manifest) ->
       if Ast_mapper.tool_name () = "ocamldep" then
         (* Just put it as manifest *)
-        { type_decl with ptype_manifest = Some manifest }
+        if is_self_reference lid then
+          { type_decl with ptype_manifest = None }
+        else
+          { type_decl with ptype_manifest = Some manifest }
       else
         with_default_loc loc (fun () ->
-          let subst = subst_of_manifest manifest in
-          let subst = subst @ [
-            `Lid (Lident (Longident.last lid)),
-              Typ.constr { txt = Lident ptype_name.txt; loc = ptype_name.loc } []
-          ] in
           let ttype_decl = locate_ttype_decl ~loc (locate_sig ~loc lid) lid in
-          let ptype_decl = ptype_decl_of_ttype_decl ~manifest ~subst ptype_name ttype_decl in
-          { ptype_decl with ptype_attributes })
+          if is_self_reference lid then begin
+            let ptype_decl = ptype_decl_of_ttype_decl ~subst:[] ptype_name ttype_decl in
+            { ptype_decl with ptype_attributes }
+          end
+          else begin
+            let subst = subst_of_manifest manifest in
+            let subst = subst @ [
+                `Lid (Lident (Longident.last lid)),
+                Typ.constr { txt = Lident ptype_name.txt; loc = ptype_name.loc } []
+              ] in
+            let ptype_decl = ptype_decl_of_ttype_decl ~manifest ~subst ptype_name ttype_decl in
```

Comment:
Done in the next push. Avoiding the duplication was my first reflex but the `?manifest` parameter had me decide otherwise to minimize the amount of change. Add also added the test, although I'm afraid it's a little minimal as the difference is whether the actual test file compiles or not.